tests/optimagic/optimization/test_optimize.py

```diff
@@ -44,3 +44,28 @@ def test_with_invalid_numdiff_options():
             algorithm="scipy_lbfgsb",
             numdiff_options={"bla": 15},
         )
+
+
+# provided fun or fun_and_jac is provided
+def test_with_optional_fun_argument():
+    expected = minimize(
+        fun=lambda x: x @ x,
+        jac=lambda x: 2 * x,
+        params=np.arange(5),
+        algorithm="scipy_lbfgsb",
+    )
+    res = minimize(
+        fun_and_jac=lambda x: (x @ x, 2 * x),
+        params=np.arange(5),
+        algorithm="scipy_lbfgsb",
+    )
+    assert np.all(res.x == expected.x) == np.True_
```

Comment:
Since we know the result of this test I would not run two optimizations (keeping test runtime short is very important). Instead I would prefer something like `expected = np.zeros(5)` and use `assert_array_almost_equal` for the comparison.